test/integration/hds_integration_test.cc

```diff
@@ -27,43 +34,28 @@ class HdsIntegrationTest : public HttpIntegrationTest,
 
   void initialize() override {
     setUpstreamCount(upstream_endpoints_);
-    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v2::Bootstrap& bootstrap) {
+    config_helper_.addConfigModifier([](envoy::config::bootstrap::v2::Bootstrap& bootstrap) {
       // Setup hds and corresponding gRPC cluster.
-      auto* hds_confid = bootstrap.mutable_hds_config();
-      hds_confid->set_api_type(envoy::api::v2::core::ApiConfigSource::GRPC);
-      hds_confid->add_grpc_services()->mutable_envoy_grpc()->set_cluster_name("hds_delegate");
+      auto* hds_config = bootstrap.mutable_hds_config();
+      hds_config->set_api_type(envoy::api::v2::core::ApiConfigSource::GRPC);
+      hds_config->add_grpc_services()->mutable_envoy_grpc()->set_cluster_name("hds_cluster");
       auto* hds_cluster = bootstrap.mutable_static_resources()->add_clusters();
       hds_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
       hds_cluster->mutable_circuit_breakers()->Clear();
-      hds_cluster->set_name("hds_delegate");
+      hds_cluster->set_name("hds_cluster");
       hds_cluster->mutable_http2_protocol_options();
-      // Switch predefined cluster_0 to EDS filesystem sourcing.
-      // TODO(lilika): Remove eds dependency
       auto* cluster_0 = bootstrap.mutable_static_resources()->mutable_clusters(0);
       cluster_0->mutable_hosts()->Clear();
-      cluster_0->set_type(envoy::api::v2::Cluster::EDS);
-      auto* eds_cluster_config = cluster_0->mutable_eds_cluster_config();
-      eds_cluster_config->mutable_eds_config()->set_path(eds_helper_.eds_path());
     });
+
     HttpIntegrationTest::initialize();
-    hds_upstream_ = fake_upstreams_[0].get();
-    for (uint32_t i = 0; i < upstream_endpoints_; ++i) {
-      service_upstream_[i] = fake_upstreams_[i + 1].get();
-    }
+    host_upstream_.reset(new FakeUpstream(0, FakeHttpConnection::Type::HTTP1, version_));
+    host2_upstream_.reset(new FakeUpstream(0, FakeHttpConnection::Type::HTTP1, version_));
```

Comment:
Technically, the `fake_upstreams_` provides a generic mechanism to allocate fake upstreams for both HDS and host upstreams, we use it this way in the load stats reporter integration test. I'm wondering if we could do something similar here? It's probably not that important, there isn't that much boilerplate here, but it would be more consistent (there is inbuilt support for `fake_upstreams_` teardown for example).